xmppserver/src/main/java/org/jivesoftware/openfire/disco/IQDiscoInfoHandler.java

```diff
@@ -701,7 +740,33 @@ public DataForm getExtendedInfo(String name, String node, JID senderJID) {
                                 continue;
                             }
                         }
+
+                        //XEP-0232 includes extended information about Software Version in a data form
+                        final DataForm dataFormSoftwareVersion = new DataForm(DataForm.Type.result);
+
+                        final FormField fieldTypeSoftwareVersion = dataFormSoftwareVersion.addField();
+                        fieldTypeSoftwareVersion.setVariable("FORM_TYPE");
+                        fieldTypeSoftwareVersion.setType(FormField.Type.hidden);
+                        fieldTypeSoftwareVersion.addValue("urn:xmpp:dataforms:softwareinfo");
+
+                        final FormField fieldOs = dataFormSoftwareVersion.addField();
+                        fieldOs.setVariable("os");
+                        fieldOs.addValue( System.getProperty("os.name"));
+
+                        final FormField fieldOsVersion = dataFormSoftwareVersion.addField();
+                        fieldOsVersion.setVariable("os_version");
+                        fieldOsVersion .addValue(System.getProperty("os.version")+" "+System.getProperty("os.arch")+" - Java " + System.getProperty("java.version"));
+
+                        final FormField fieldSoftware = dataFormSoftwareVersion.addField();
+                        fieldSoftware.setVariable("software");
+                        fieldSoftware.addValue(AdminConsole.getAppName());
+
+                        final FormField fieldSoftwareVersion = dataFormSoftwareVersion.addField();
+                        fieldSoftwareVersion.setVariable("software_version");
+                        fieldSoftwareVersion.addValue(AdminConsole.getVersionString());
+
                         final Set<DataForm> dataForms = new HashSet<>();
+                        dataForms.add(dataFormSoftwareVersion);
```

Comment:
Whenever a new feature is built, it is good to include a "kill-switch": a property that, if changed, disables the functionality. This allows people to work around any issues that are introduced by the new feature (if any).

Please include a new `SystemProperty` of type Boolean, that by default is true, that is used to see if the new XEP-0232 dataform should be created/returned.